projects/models/project.py

```diff
@@ -58,15 +55,37 @@ def __str__(self):
         return self.name
 
 
-class ProjectPhaseAttribute(models.Model):
+class ProjectPhaseSection(models.Model):
+    phase = models.ForeignKey(ProjectPhase, verbose_name=_('phase'), related_name='sections', on_delete=models.CASCADE)
+    name = models.CharField(max_length=255, verbose_name=_('name'))
+    index = models.PositiveIntegerField(verbose_name=_('index'), null=True, blank=True)
+    attributes = models.ManyToManyField(
+        Attribute, verbose_name=_('attributes'), related_name='phase_sections', through='ProjectPhaseSectionAttribute'
+    )
+
+    class Meta:
+        verbose_name = _('project phase section')
+        verbose_name_plural = _('project phase sections')
+        unique_together = ('phase', 'index')
+        ordering = ('phase', 'index')
+
+    def __str__(self):
+        return self.name
+
+    def get_attribute_identifiers(self):
+        return [a.identifier for a in self.attributes.all()]
+
+
+class ProjectPhaseSectionAttribute(models.Model):
     attribute = models.ForeignKey(Attribute, verbose_name=_('attribute'), on_delete=models.CASCADE)
-    phase = models.ForeignKey(ProjectPhase, verbose_name=_('phase'), on_delete=models.CASCADE)
+    section = models.ForeignKey(ProjectPhaseSection, verbose_name=_('phase section'), on_delete=models.CASCADE)
+    generated = models.BooleanField(verbose_name=_('generated'))
     required = models.BooleanField(verbose_name=_('required'))
     index = models.PositiveIntegerField(verbose_name=_('index'), null=True, blank=True)
```

Comment:
The same as the section index, and actually all indexes, no reason to make them nullable.